src/main/java/Tasks/Task.java

```diff
@@ -0,0 +1,39 @@
+package tasks;
+
+public abstract class Task {
+    protected String name;
+    protected Boolean isDone;
+
+    public Task(String name) {
+        this.name = name;
+        this.isDone = false;
+    }
+
+    /**
+     * Toggles isDone as true.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Toggles isDone as true.
+     */
+    public void markAsNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns string representation of the isDone status.
+     */
+    public String getTaskStatus() {
+        return (isDone ? "X" : " ");
+    }
+
+    public abstract String getWriteTaskInfo();
```

Comment:
Since this is an abstract method (which needs to be implemented by subclasses), would it be better to include some javadoc on what should the output look like? (or maybe what the method should do?)